### DISCUSSION TITLE
How to update ChatClient#getCurrentUser instance after changing user mutes?

### DISCUSSION THREAD

**Discussion by rupinderjeet** [GetStream/stream-chat-android#1686] [Q&A]
From docs, user mutes and channel mutes are only received as a list when I connect a user. After that if I change these mutes, they are received in notification events i.e. `NotificationMutesUpdatedEvent`, `NotificationChannelMutesUpdatedEvent`. But, these changes aren't reflected in `ChatClient.instance().getCurrentUser()`. Doesn't it make this instance of current-user unreliable?

Is this intentional? If so, only way to handle it should be some sort of database. Have I understood this correctly?

Also, is there any limit on how many users/channels I can mute? If I mute 1000 users, will I get them all in the `user` returned by `ChatClient#connectUser()` method? 

And, can I rely on `NotificationMutesUpdatedEvent` event to have list of all the mutes? In this case, I will just replace all db entries of mutes I have locally available.

**Comment by JcMinarro** [comment]
Hello @rupinderjeet 

If you are using only our Low Level Client (`stream-chat-android-client`) the currentUser inside of `ChatClient` is not updated by the event you receive.
You will need to keep you data updated by subscribing to ChatEvents and handling [those events](https://github.com/GetStream/stream-chat-android/blob/main/stream-chat-android-client/src/main/java/io/getstream/chat/android/client/events/ChatEvent.kt)

If you are using our Offline Support SDK (`stream-chat-android-offline`), you can check `ChatDomain::currentUser` where we keep data sync

**Comment by rupinderjeet** [comment]
Ah, I was missing this then. I will try it tomorrow morning, thanks.

**Comment by tschellenbach** [comment]
There is also chatDomain.muted: https://getstream.io/chat/docs/android/android_offline/?language=kotlin#building-the-chatdomain

**Comment by rupinderjeet** [comment]
Hello @JcMinarro @tschellenbach 

Is there any upper limit on how many users/channels I can mute? If I mute 1000 users, will I get them all in the user returned by `ChatDomain.instance().currentUser` method?

My current usecase is just that I have to check if a target user is muted by me so that I can show proper options UI for user-profile. So, I think observing `muted` livedata is not applicable here. If there is no such limit on user-mutes, shouldn't we have something like 

```
val isMuted = ChatDomain.instance().isUserMuted(userId)
val isChannelMuted = ChatDomain.instance().isChannelMuted(channelId)
```

I am just worried that carrying limitless user-instances for mutes in currentUser can lead to some kind of overhead. If I am wrong, then I guess I will start using it as it is.
